Sources/BraintreeCoreSwift/BTAPIClientSwift.swift

```diff
@@ -1,5 +1,440 @@
+
+            if error != nil {
+                completion(nil, error)
+                return
+            } else if response?.statusCode != 200 {
+                completion(nil, BTAPIClientError.configurationUnavailable)
+                return
+            } else {
+                configuration = BTConfiguration(json: body)
+
+                if self.apiHTTP == nil {
+                    let apiURL: URL? = configuration?.json?["clientApiUrl"].asURL()
+                    let accessToken: String? = configuration?.json?["braintreeApi"]["accessToken"].asString()
+
+                    if let apiURL, let accessToken {
+                        self.apiHTTP = BTAPIHTTP(url: apiURL, accessToken: accessToken)
+                    }
+                }
+
+                if self.http == nil {
+                    let baseURL: URL? = configuration?.json?["clientApiUrl"].asURL()
+
+                    if let clientToken = self.clientToken, let baseURL {
+                        self.http = BTHTTP(url: baseURL, authorizationFingerprint: clientToken.authorizationFingerprint)
+                    } else if let tokenizationKey = self.tokenizationKey, let baseURL {
+                        self.http = BTHTTP(url: baseURL, tokenizationKey: tokenizationKey)
+                    }
+                }
+
+                if self.graphQLHTTP == nil {
+                    let graphQLBaseURL: URL? = self.graphQLURL(forEnvironment: configuration?.environment ?? "")
+
+                    if let clientToken = self.clientToken, let graphQLBaseURL {
+                        self.graphQLHTTP = BTGraphQLHTTP(url: graphQLBaseURL, authorizationFingerprint: clientToken.authorizationFingerprint)
+                    } else if let tokenizationKey = self.tokenizationKey, let graphQLBaseURL {
+                        self.graphQLHTTP = BTGraphQLHTTP(url: graphQLBaseURL, tokenizationKey: tokenizationKey)
+                    }
+                }
+            }
+
+            completion(configuration, nil)
+        }
+    }
+
+    /// Fetches a customer's vaulted payment method nonces.
+    /// Must be using client token with a customer ID specified.
+    ///  - Parameter completion: Callback that returns either an array of payment method nonces or an error
+    public func fetchPaymentMethodNonces(completion: @escaping ([BTPaymentMethodNonce]?, Error?) -> Void) {
+        fetchPaymentMethodNonces(false, completion: completion)
+    }
+
+    /// Fetches a customer's vaulted payment method nonces.
+    /// Must be using client token with a customer ID specified.
+    ///  - Parameters:
+    ///   - defaultFirst: Specifies whether to sort the fetched payment method nonces with the default payment method or the most recently used payment method first
+    ///   - completion: Callback that returns either an array of payment method nonces or an error
+    public func fetchPaymentMethodNonces(_ defaultFirst: Bool, completion: @escaping ([BTPaymentMethodNonce]?, Error?) -> Void) {
```

Comment:
In the Objective-C file we have a TODO to move this function to the Drop-in. That said, with the Drop-in getting less love these days I could see this function being useful for core merchants. Curious to hear other thoughts!

Reply:
Yeah we moved this to DropIn because of circular dependencies on Android but we did get an inbound for a merchant asking to bring it back. If a merch wanted to build their own DropIn like UI it would be useful.

Reply:
Yeah, even thinking for building a UI in general would be a nice helper function to have in the SDK.